### Build Failed!

The last (useful) 30 log lines are shown below:
[View Full Log](https://console.aws.amazon.com/cloudwatch/home?region=ap-southeast-2#logEvent:group=/aws/codebuild/assignar-ci;stream=0d5ddd5c-cd84-42a8-8f0f-53502ffd9607)
```
Cloning into 'eslint-config-assignar'...


node version: v8.10.0
yarn version: 1.5.1


yarn install v1.5.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 8.74s.


$ make ci
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-base/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-graphql/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-node/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-react/tsconfig.json
$ eslint --cache ./packages --ext .ts
# show-unused
base...

unused rules

jest/expect-expect                      jest/no-test-return-statement
jest/prefer-inline-snapshots            jest/prefer-strict-equal
no-async-promise-executor               no-misleading-character-class
require-atomic-updates                  require-unicode-regexp
```
<details><summary>Unfiltered log</summary>
<p>

```
[Container] 2018/08/24 02:40:03 Waiting for agent ping
[Container] 2018/08/24 02:40:03 Waiting for DOWNLOAD_SOURCE
[Container] 2018/08/24 02:40:03 Phase is DOWNLOAD_SOURCE
[Container] 2018/08/24 02:40:03 CODEBUILD_SRC_DIR=/codebuild/output/src050753314/src/github.com/assignar/ci
[Container] 2018/08/24 02:40:03 YAML location is /codebuild/output/src050753314/src/github.com/assignar/ci/buildspec.yml
[Container] 2018/08/24 02:40:03 Processing environment variables
[Container] 2018/08/24 02:40:03 Decrypting parameter store environment variables
[Container] 2018/08/24 02:40:03 Moving to directory /codebuild/output/src050753314/src/github.com/assignar/ci
[Container] 2018/08/24 02:40:03 Registering with agent
[Container] 2018/08/24 02:40:03 Phases found in YAML: 1
[Container] 2018/08/24 02:40:03  BUILD: 1 commands
[Container] 2018/08/24 02:40:03 Phase complete: DOWNLOAD_SOURCE Success: true
[Container] 2018/08/24 02:40:03 Phase context status code:  Message:
[Container] 2018/08/24 02:40:03 Entering phase INSTALL
[Container] 2018/08/24 02:40:03 Phase complete: INSTALL Success: true
[Container] 2018/08/24 02:40:03 Phase context status code:  Message:
[Container] 2018/08/24 02:40:03 Entering phase PRE_BUILD
[Container] 2018/08/24 02:40:03 Phase complete: PRE_BUILD Success: true
[Container] 2018/08/24 02:40:03 Phase context status code:  Message:
[Container] 2018/08/24 02:40:03 Entering phase BUILD
[Container] 2018/08/24 02:40:03 Running command node ./test.js
Cloning into 'eslint-config-assignar'...
Note: checking out '5803326816b1fba7d058f1aca798f524b394611b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 5803326... [DEP] update deps


node version: v8.10.0
yarn version: 1.5.1


yarn install v1.5.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 8.74s.


yarn run v1.5.1
$ make ci
make[1]: Entering directory '/tmp/eslint-config-assignar'
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-base/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-graphql/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-node/tsconfig.json
$ /tmp/eslint-config-assignar/node_modules/.bin/tsc -p ./packages/eslint-config-assignar-react/tsconfig.json
make[1]: Leaving directory '/tmp/eslint-config-assignar'
make[1]: Entering directory '/tmp/eslint-config-assignar'
$ eslint --cache ./packages --ext .ts
make[1]: Leaving directory '/tmp/eslint-config-assignar'
# show-unused
make[1]: Entering directory '/tmp/eslint-config-assignar'
base...

unused rules

jest/expect-expect                      jest/no-test-return-statement
jest/prefer-inline-snapshots            jest/prefer-strict-equal
no-async-promise-executor               no-misleading-character-class
require-atomic-updates                  require-unicode-regexp
Makefile:13: recipe for target 'show-unused' failed
make[1]: *** [show-unused] Error 1
make[1]: Leaving directory '/tmp/eslint-config-assignar'
Makefile:30: recipe for target 'ci' failed
make: *** [ci] Error 2
error An unexpected error occurred: "Command failed.
Exit code: 2
Command: sh
Arguments: -c make ci
Directory: /tmp/eslint-config-assignar
Output:
".
info If you think this is a bug, please open a bug report with the information provided in "/tmp/eslint-config-assignar/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
child_process.js:644
    throw err;
    ^

Error: Command failed: yarn ci
    at checkExecSyncError (child_process.js:601:13)
    at Object.execSync (child_process.js:641:13)
    at execSync (/codebuild/output/src050753314/src/github.com/assignar/ci/test.js:7:18)
    at Object.<anonymous> (/codebuild/output/src050753314/src/github.com/assignar/ci/test.js:108:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)

[Container] 2018/08/24 02:40:28 Command did not exit successfully node ./test.js exit status 1
[Container] 2018/08/24 02:40:28 Phase complete: BUILD Success: false
[Container] 2018/08/24 02:40:28 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: node ./test.js. Reason: exit status 1
[Container] 2018/08/24 02:40:29 Entering phase POST_BUILD
[Container] 2018/08/24 02:40:29 Phase complete: POST_BUILD Success: true
[Container] 2018/08/24 02:40:29 Phase context status code:  Message:
```

</p>
</details>

[Click here to retry the build](https://ut2fgw3cae.execute-api.ap-southeast-2.amazonaws.com/development//retry-build?repo=eslint-config-assignar&commit=5803326816b1fba7d058f1aca798f524b394611b&retryToken=8dbe9aaa-d208-4403-975d-2b9c84238305)

--- a/package.json
+++ b/package.json
@@ -7,18 +7,18 @@
     "author": "Brad Zacher <brad.zacher@gmail.com>",
     "license": "MIT",
     "devDependencies": {
-        "@assignar/api-schemas": "^5.0.2",
+        "@assignar/api-schemas": "^5.0.21",
         "@types/eslint": "^4.16.3",
         "@types/graphql": "^0.13.3",
         "@types/node": "^10.5.1",
-        "eslint": "^5.0.1",
+        "eslint": "^5.4.0",
         "eslint-find-rules": "^3.3.1",
         "eslint-import-resolver-node": "^0.3.2",
         "graphql": "^0.13.2",
         "lerna": "^2.11.0",
         "rimraf": "^2.6.2",
-        "ts-node": "^7.0.0",
-        "typescript": "^2.9.2"
+        "ts-node": "^7.0.1",
+        "typescript": "^3.0.1"
     },
     "scripts": {
         "typecheck": "tsc -p ./tsconfig.json --noEmit",

--- a/package.json
+++ b/package.json
@@ -9,8 +9,8 @@
     "devDependencies": {
         "@assignar/api-schemas": "^5.0.21",
         "@types/eslint": "^4.16.3",
-        "@types/graphql": "^0.13.3",
-        "@types/node": "^10.5.1",
+        "@types/graphql": "^0.13.4",
+        "@types/node": "^10.9.1",
         "eslint": "^5.4.0",
         "eslint-find-rules": "^3.3.1",
         "eslint-import-resolver-node": "^0.3.2",

--- a/packages/eslint-config-assignar-base/package.json
+++ b/packages/eslint-config-assignar-base/package.json
@@ -6,14 +6,14 @@
     "author": "Brad Zacher <brad.zacher@gmail.com>",
     "main": "./dist/index.js",
     "dependencies": {
-        "eslint": "^5.0.1",
+        "eslint": "^5.4.0",
         "eslint-plugin-dependencies": "^2.4.0",
         "eslint-plugin-eslint-comments": "^3.0.1",
-        "eslint-plugin-import": "^2.13.0",
-        "eslint-plugin-jest": "^21.15.0",
+        "eslint-plugin-import": "^2.14.0",
+        "eslint-plugin-jest": "^21.21.0",
         "eslint-plugin-typescript": "^0.12.0",
         "eslint-restricted-globals": "^0.1.1",
-        "typescript-eslint-parser": "^16.0.1"
+        "typescript-eslint-parser": "^18.0.0"
     },
     "peerDependencies": {
         "typescript": "*"

--- a/packages/eslint-config-assignar-base/src/plugins/eslint-base.ts
+++ b/packages/eslint-config-assignar-base/src/plugins/eslint-base.ts
@@ -548,6 +548,10 @@ const rules : Rules.Eslint = {
     // https://eslint.org/docs/rules/no-array-constructor
     'no-array-constructor': 'error',
 
+    // The new Promise constructor accepts an executor function as an argument, which has resolve and reject parameters that can be used to control the state of the created Promise
+    // https://eslint.org/docs/rules/no-async-promise-executor
+    'no-async-promise-executor': 'error',
+
     // This rule disallows the use of await within loop bodies.
     // https://eslint.org/docs/rules/no-await-in-loop
     'no-await-in-loop': 'error',
@@ -815,6 +819,10 @@ const rules : Rules.Eslint = {
     // This rule aims to make code more readable by ensuring that special numbers are declared as constants
     // https://eslint.org/docs/rules/no-magic-numbers
     'no-magic-numbers': 'off',
+
+    // Disallow characters which are made with multiple code points in character class syntax, basically disallows emojis to be used as class names
+    // https://eslint.org/docs/rules/no-misleading-character-class
+    'no-misleading-character-class': 'error',
 
     // Disallow mixes of different operators
     // https://eslint.org/docs/rules/no-mixed-operators
@@ -1477,12 +1485,20 @@ const rules : Rules.Eslint = {
     radix: 'error',
 
     // This rule warns async functions which have no await expression
+    // https://eslint.org/docs/rules/require-atomic-updates
+    'require-atomic-updates': 'error',
+
+    // This rule warns async functions which have no await expression
     // https://eslint.org/docs/rules/require-await
     'require-await': 'error',
 
     // This rule requires JSDoc comments for specified nodes
     // https://eslint.org/docs/rules/require-jsdoc
     'require-jsdoc': 'off',
+
+    // Enforce the use of u flag on RegExp
+    // https://eslint.org/docs/rules/require-unicode-regexp
+    'require-unicode-regexp': 'off',
 
     // This rule generates warnings for generator functions that do not have the yield keyword.
     // https://eslint.org/docs/rules/require-yield

--- a/packages/eslint-config-assignar-base/src/plugins/jest.ts
+++ b/packages/eslint-config-assignar-base/src/plugins/jest.ts
@@ -9,6 +9,10 @@ const rules : Rules.Jest = {
         },
     ],
 
+    // Ensure that there is at least one expect call made in a test
+    // https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/expect-expect.md
+    'jest/expect-expect': 'error',
+
     // Enforce it, test and describe to have descriptions that begin with a lowercase letter.
     // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/lowercase-name.md
     'jest/lowercase-name': [
@@ -50,10 +54,22 @@ const rules : Rules.Jest = {
     // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-test-prefixes.md
     'jest/no-test-prefixes': 'error',
 
+    // Disallow explicitly returning from tests
+    // https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/no-test-return-statement.md
+    'jest/no-test-return-statement': 'error',
+
     // Suggest using expect.assertions() OR expect.hasAssertions()
     // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-expect-assertions.md
     // this rule adds no real value
     'jest/prefer-expect-assertions': 'off',
+
+    // Suggest using inline snapshots
+    // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-inline-snapshots.md
+    'jest/prefer-inline-snapshots': 'off',
+
+    // Suggest using toStrictEqual()
+    // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-strict-equal.md
+    'jest/prefer-strict-equal': 'off',
 
     // Suggest using toBeNull()
     // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-to-be-null.md

--- a/packages/eslint-config-assignar-react/package.json
+++ b/packages/eslint-config-assignar-react/package.json
@@ -9,8 +9,5 @@
         "eslint-config-assignar-base": "^1.2.1",
         "eslint-plugin-react": "^7.11.1"
     },
-    "devDependencies": {
-        "@types/graphql": "^0.13.4",
-        "@types/node": "^10.9.1"
-    }
+    "devDependencies": {}
 }

--- a/packages/eslint-config-assignar-react/package.json
+++ b/packages/eslint-config-assignar-react/package.json
@@ -7,6 +7,10 @@
     "main": "./dist/index.js",
     "dependencies": {
         "eslint-config-assignar-base": "^1.2.1",
-        "eslint-plugin-react": "^7.10.0"
+        "eslint-plugin-react": "^7.11.1"
+    },
+    "devDependencies": {
+        "@types/graphql": "^0.13.4",
+        "@types/node": "^10.9.1"
     }
 }

--- a/types/eslint-config.d.ts
+++ b/types/eslint-config.d.ts
@@ -75,6 +75,7 @@ declare namespace Rules {
          'new-parens' : RuleType
          'newline-per-chained-call' : RuleType
          'no-alert' : RuleType
+         'no-async-promise-executor' : RuleType
          'no-array-constructor' : RuleType
          'no-await-in-loop' : RuleType
          'no-bitwise' : RuleType
@@ -132,6 +133,7 @@ declare namespace Rules {
          'no-lonely-if' : RuleType
          'no-loop-func' : RuleType
          'no-magic-numbers' : RuleType
+         'no-misleading-character-class' : RuleType
          'no-mixed-operators' : RuleType
          'no-mixed-requires' : RuleType
          'no-mixed-spaces-and-tabs' : RuleType
@@ -229,8 +231,10 @@ declare namespace Rules {
          'quote-props' : RuleType
          'quotes' : RuleType
          'radix' : RuleType
+         'require-atomic-updates' : RuleType
          'require-await' : RuleType
          'require-jsdoc' : RuleType
+         'require-unicode-regexp' : RuleType
          'require-yield' : RuleType
          'rest-spread-spacing' : RuleType
          'semi' : RuleType
@@ -318,23 +322,27 @@ declare namespace Rules {
          'import/unambiguous' : RuleType
     }
     export interface Jest {
-         'jest/consistent-test-it' : RuleType
-         'jest/lowercase-name' : RuleType
-         'jest/no-disabled-tests' : RuleType
-         'jest/no-focused-tests' : RuleType
-         'jest/no-hooks' : RuleType
-         'jest/no-identical-title' : RuleType
-         'jest/no-jasmine-globals' : RuleType
-         'jest/no-jest-import' : RuleType
-         'jest/no-large-snapshots' : RuleType
-         'jest/no-test-prefixes' : RuleType
-         'jest/prefer-expect-assertions' : RuleType
-         'jest/prefer-to-be-null' : RuleType
-         'jest/prefer-to-be-undefined' : RuleType
-         'jest/prefer-to-have-length' : RuleType
-         'jest/valid-describe' : RuleType
-         'jest/valid-expect' : RuleType
-         'jest/valid-expect-in-promise' : RuleType
+        'jest/consistent-test-it' : RuleType
+        'jest/expect-expect' : RuleType
+        'jest/lowercase-name' : RuleType
+        'jest/no-disabled-tests' : RuleType
+        'jest/no-focused-tests' : RuleType
+        'jest/no-hooks' : RuleType
+        'jest/no-identical-title' : RuleType
+        'jest/no-jasmine-globals' : RuleType
+        'jest/no-jest-import' : RuleType
+        'jest/no-large-snapshots' : RuleType
+        'jest/no-test-prefixes' : RuleType
+        'jest/no-test-return-statement' : RuleType
+        'jest/prefer-expect-assertions' : RuleType
+        'jest/prefer-inline-snapshots' : RuleType
+        'jest/prefer-strict-equal' : RuleType
+        'jest/prefer-to-be-null' : RuleType
+        'jest/prefer-to-be-undefined' : RuleType
+        'jest/prefer-to-have-length' : RuleType
+        'jest/valid-describe' : RuleType
+        'jest/valid-expect' : RuleType
+        'jest/valid-expect-in-promise' : RuleType
     }
     export interface React {
          'react/boolean-prop-naming' : RuleType

